### PR TITLE
SierraBibRetrievalRequest

### PR DESCRIPTION
**Pull request by jobinthomasnypl**
Retrieve bibs and post to SierraBibPostRequest stream. This is a first pass. Please feel free to critique to make sure this follows our dev standards and best practices.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,33 +1,96 @@
...
+  var verify = type.fromBuffer(decoded);
+  return JSON.parse(verify);
+}
 
-  var id = crypto.randomBytes(20).toString('hex').toString()
+//get full bib details for each bib
+var loadedConfig = wrapper.loadConfig('config.json');
+var bib_in_detail = function(bibId) {
+  return new Promise(function (resolve, reject) {
+    wrapper.auth((error, results) => {
```
Comment (EdwinGuzman) [inline index.js]: Is it possible to authenticate only once and then call the API for a bib instead of trying to authenticate for every bib request?

**Comment by EdwinGuzman** [comment]
Looks like a good first pass, Jobin. The next pass an include "best practices". Maybe we can all agree on an eslint convention to use. Make sure for the next step, you store the schema in memory (locally) so you don't call for the schema for every bib.

File: index.js (hunk 1 above)
Comment (jobinthomasnypl) [inline index.js]: Hi @EdwinGuzman,

I was looking at the documentation for the sierra-wrapper. It seems like the wrapper is storing the authentication token and renews it when it expires. 

https://www.npmjs.com/package/sierra-wrapper#auth

https://github.com/NYPL-discovery/sierra-wrapper/blob/master/index.js (lines 70-110)
@thisismattmiller 

We can look at it together.

**Comment by EdwinGuzman** [comment]
Can you also write a bit of documentation on how to run it? An empty `config.json` file can include too:

```
{
  "key": "",
  "secret": "",
  "base": ""
}
```

**Comment by jobinthomasnypl** [comment]
@EdwinGuzman,

When you get a chance can you review the latest changes I added for `	Externalize values in config.json file`.

I am getting a warning - `(node:52538) DeprecationWarning: Calling an asynchronous function without callback is deprecated.`

**Comment by EdwinGuzman** [comment]
Works well @jobinthomasnypl . Just commit a `context.json` file with an empty object `{}`. I think that should be enough for this PR. You can update the README and the `wrapper.auth()` code in a new PR.

**Comment by jobinthomasnypl** [comment]
Updated readme. That should handle most of the feedback in the PR. I will add standardjs for code formatting.

**Comment by jobinthomasnypl** [comment]
This PR can be closed and the other one can be used for further feedback and comments.

**Pull request closed without merging**